clients/src/main/java/org/apache/kafka/common/requests/ControlledShutdownRequest.java

```diff
@@ -60,10 +80,12 @@ public int brokerId() {
     }
 
     public static ControlledShutdownRequest parse(ByteBuffer buffer, int versionId) {
```

Comment:
I wanted to do this, but it got messy.  The problem is that there are a bunch of unit tests that are using reflection to call these methods-- and also some scala code.  I think we should have a follow-on JIRA for converting the parse() methods to use 'short' for version.